JM/CF/Scripts/3_Game/CommunityFramework/ModStorage/CF_ModStorageModule.c

```diff
@@ -0,0 +1,165 @@
+/**
+ * @class CF_ModStorageModule
+ * 
+ * @brief Does not support unloading CF from the mods. Once loaded it can't be removed
+ */
+[CF_RegisterModule(CF_ModStorageModule)]
+class CF_ModStorageModule : CF_ModuleGame
+{
+	protected string m_FilePath;
+
+	protected autoptr map<int, ref map<int, ref map<int, ref map<int, ref Param1<bool>>>>> m_IDs = new map<int, ref map<int, ref map<int, ref map<int, ref Param1<bool>>>>>();
+	protected bool m_IsLoaded;
+
+	protected autoptr FileSerializer m_Serializer;
+
+	/**
+	 * @brief Checks if the persistent ID is in the map. If they aren't in the map then add and write to the file
+	 */
+	void AddEntity(int b1, int b2, int b3, int b4)
+	{
+		Load();
+
+		if (!_AddEntity(b1, b2, b3, b4, false))
+		{
+			return;
+		}
+
+		m_Serializer.Write(b1);
+		m_Serializer.Write(b2);
+		m_Serializer.Write(b3);
+		m_Serializer.Write(b4);
+	}
+
+	/**
+	 * @brief Checks to see if the persistent ID is loaded
+	 */
+	bool IsEntity(int b1, int b2, int b3, int b4)
+	{
+		Load();
+
+		auto map_b1 =  m_IDs[b1];
+		if (!map_b1)
+		{
+			return false;
+		}
+
+		auto map_b2 = map_b1[b2];
+		if (!map_b2)
+		{
+			return false;
+		}
+
+		auto map_b3 = map_b2[b3];
+		if (!map_b3)
+		{
+			return false;
+		}
+
+		auto val_b4 = map_b3[b4];
+		if (!val_b4)
+		{
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * @brief Reads the modstorage file
+	 */
+	void Load(bool reload = false)
+	{
+		if (m_IsLoaded && !reload)
+		{
+			return;
+		}
+		
+		m_IsLoaded = true;
+
+		int instanceId = g_Game.ServerConfigGetInt("instanceId");
+
+		string folder = "$mission:storage_" + instanceId + "/";
+		if (!FileExist(folder))
+		{
+			MakeDirectory(folder);
+		}
+
+		folder += "communityframework/";
+		if (!FileExist(folder))
+		{
+			MakeDirectory(folder);
+		}
+
+		m_FilePath = folder + "cf_modstoragemodule.bin";
```

Comment:
I find the name is a bit odd. sure we prefix our classes etc, but i think this file could have a nicer name such as `modstoragemeta.bin` or `modstorageids.bin` or simply `modstorage.bin`

Reply:
Can change. Was holdover from before the file was generated inside of the `communityframework` folder.